src/spec.ts

```diff
@@ -50,8 +50,27 @@ export interface FacetSpec extends BaseSpec {
   spec: LayerSpec | UnitSpec;
 }
 
+export enum ResolveStrategy {
+  INDEPENDENT = 'independent' as any,
+  SHARED = 'shared' as any
+}
+
+export const INDEPENDENT = ResolveStrategy.INDEPENDENT;
+export const SHARED = ResolveStrategy.SHARED;
+
+export interface Resolve {
+  scale: ResolveStrategy;
+  guide?: ResolveStrategy;
```

Comment:
The rationale is that we don't have to teach that  guide collective means these two terms because practically guide also include annotations (in addition to `axis` and `legend`)